lamkey_nick_ndvi.py

```diff
@@ -0,0 +1,723 @@
+    single_scene_points += 2
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_ts_single_site.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    single_scene_points += 2
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Ensure the site name is correct
+if student_ndvi_ts_single_site.site.values[0] == 'HARV':
+    print('\u2705 You have the correct site name!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct site name.')
+
+if np.allclose(0.281131628228094, student_ndvi_ts_single_site.mean_ndvi.values[0]):
+    print('\u2705 You have the correct mean NDVI value!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct mean ndvi value.')
+
+print("\n \u27A1 You received {} out of 15 points for creating a dataframe.".format(
+    single_scene_points))
+single_scene_points
+
+
+# ## Task 2:
+# 
+# In the cell below, process all of the landsat scenes. Create a DataFrame that contains the following 
+# information for each scene
+# 
+# 
+# |   | index  | site  | mean_ndvi  | 
+# |---|---|---|---|
+# | Date  |   |   |   |
+# |  2017-01-07  | 0  | SJER  | .4  |  
+# 
+# Be sure to call your dataframe at the end of the cell to ensure autograding works.
+# HINT: FOR THIS STEP, leave any rows containing missing values (`NAN`).
+
+# # Processing multiple scenes from multiple sites 
+# 
+# ### What's the workflow?
+# * Below the same process used above is also used here in the to process multiple sites. The master for loop iterates through two folders. As the master loop iterates a landsat_dir list is generated for a site which is processed in the subsequent for loops the same way as above but now the master for loop will go through both directories containing each site. As long as the landsat imagery is stored in a folder labeled sites with vector layers are named with site_name + -crop.shp, this loop will iterate through any number of sites and scenes. 
+# 
+# ### How was the code optimized?
+# * The loop structure of this code keeps track of each scene so that only the bands from one scene are being worked on through each iteration. Only the date, site names and the final ndvi values are stored outside the loop. The two functions also eliminate variables being stored in memory. Pyproj was used to speed up processing times. 
+
+# In[8]:
+
+
+# Create dataframe of NDVI including the cleaning data to deal with clouds
+
+# Important: to use the ungraded tests below as a sanity check,
+# name your columns: mean_ndvi and site
+# Don't forget to set date as the index and make the values of type datetime
+
+# Path to data
+path_to_data = os.path.join("ndvi-automation", "sites")
+
+# Get list of both site directories
+site_dirs = glob(path_to_data + "/*/")
```

Comment:
@nlamkey  be cautious of using delimiters within strings, as this could potentially cause issues b/w different os's

Reply:
Thanks for the heads up, I did not know that.